Sources/SwiftProtobuf/TextFormatScanner.swift

```diff
@@ -128,7 +128,11 @@ private func decodeString(_ s: String) -> String? {
             if let digit3 = bytes.next(),
               digit3 >= asciiZero && digit3 <= asciiSeven {
               let digit3Value = digit3 - asciiZero
-              let n = digit1Value * 64 + digit2Value * 8 + digit3Value
+              // The max octal digit is actually \377, but looking at the C++
+              // protobuf code in strutil.cc:UnescapeCEscapeSequences(), it
+              // decodes with rollover, so just duplicate that behavior for
+              // consistency between languages.
+              let n = digit1Value &* 64 &+ digit2Value &* 8 &+ digit3Value
```

Comment:
That's interesting.   Hmmm....  We need to get the protobuf folks to cobble up a conformance tester for TextFormat, I suppose.  We also should have a test case for this.

Reply:
(Or feel free to just commit this and I'll fill in tests later...)

Reply:
I'm looking at doing a follow up to make it easier to make test cases out of stuff the fuzzer finds, so I'll try to push on that next and bring back some of the fixes from them.